src/unk_0203E880.c

```diff
@@ -674,12 +674,12 @@ void sub_0203F21C (FieldSystem * fieldSystem, u16 param1, u16 param2, u16 param3
     *(u16 *)sub_0203F098(fieldSystem, 44) = param4;
 }
 
-u16 sub_0203F254 (u16 param0)
+u16 Script_GetTrainerIDFromEventID(u16 eventID)
 {
-    if (param0 < 5000) {
-        return param0 - 3000 + 1;
+    if (eventID < 5000) {
+        return eventID - 3000 + 1;
     } else {
-        return param0 - 5000 + 1;
+        return eventID - 5000 + 1;
     }
 }
```

Comment:
Two things:

1. Does this match as a ternary?

```c
return eventID < 5000 ? eventID - 3000 + 1 : eventID - 5000 + 1;
```

2. Script IDs in the range `[3000,5000)` correspond to single battles, and those in the range `[5000,7000)` correspond to double battles. Ergo, I think that these "magic" values could do with some named constants.

```c
#define SINGLE_BATTLE_EVENT_ID_OFFSET 3000
#define DOUBLE_BATTLE_EVENT_ID_OFFSET 5000
```

Reply:
Ternary doesn't match